report/2.Analysis/ArduinoLanguagesCtrl.tex

```diff
@@ -1,70 +1,130 @@
+ * board blink on and off every 500 milliseconds
  */
 
-const int PIN13 = 13;
+#include "Arduino.h" //The Arduino standard library
+#include "Blinker.h" //The Blinker custom library
+
+#define LED_PIN 13 //Symbolic constant
+
+const int BLINK_DELAY = 250; //Constant, global integer variable
+
+Blinker blinker(LED_PIN, BLINK_DELAY); //Object instantiation
+
+void setup() {} //No setup code for this program
+
+void loop()
+{
+    blinker.Blink(); //Program blinks infinitely
+}
 
-void setup() {
-    // Setup code
-    pinMode(PIN13, OUTPUT); // Set digital pin 13 to output
+Blinker::Blinker(int pin, int msDelay)
+{
+	pinMode(pin, OUTPUT);
+	_pin = pin;
+	_msDelay = msDelay;
 }
 
-void loop() {
-    // Loop code is looped infinitely
-    digitalWrite(PIN13, HIGH); // Set output for pin 13 to high
-    delay(500);                // Delay for 500 milliseconds
-    digitalWrite(PIN13, LOW);  // Set output for pin 13 to low
-    delay(500);                // Delay for 500 milliseconds
+void Blinker::Blink()
+{
+	digitalWrite(_pin, HIGH);
+	delay(_msDelay);
+	digitalWrite(_pin, LOW);
+	delay(_msDelay);
 }
 
 \end{lstlisting}
 
+\begin{lstlisting}[language=C++,label=lis:arduinoHeader,caption=The Blinker.h header file.,firstnumber=1]
+class Blinker
+{
+	public:
+		Blinker(int pin, int msDelay); //Constructor
+		void Blink();
+	private:
+		int _pin;
+		int _msDelay;
+};
+\end{lstlisting}
+
 % Source: concurrency.cc
 \textbf{Occam-pi} is a language that focuses on making parallel programming easy for programmers.
 It is based on Occam that first appeared in 1983.
-The full documentation of Occam-pi can be read on the official website of the language, \url{concurrency.cc}, but here is a short description of some of the syntax.
+The full documentation of Occam-pi can be read on the official website of the language, \url{concurrency.cc}, but following here is a short description of some of the syntax, and an example program in Snippet \ref{lis:occamExample}.
 
-Functions in Occam-pi are declared using the keyword \texttt{PROC} followed by a name and the function parameters enclosed in parentheses.
+\textit{Variables} are declared by initializing them with the colon-equals operator (lines 6-7). Type is automatically inferred by the expression on the right hand side of the initializing assignment.
+
+\textit{Functions} in Occam-pi are declared using the keyword \texttt{PROC} followed by a name and the function parameters enclosed in parentheses (line 5).
 Unlike C, Occam-pi doesn't use curly-brackets to encapsulate the body.
 Instead, the body is indented using a tab, similar to the Python language.
 The end of the function is marked using a colon.
 One of the key features of Occam-pi is the fact that it separates expressions that are evaluated sequentially and ones that are evaluated concurrently using the \texttt{SEQ} and \texttt{PAR} keywords.
-Comments are prefixed with \texttt{"{-}{-}"}.
 
-A typical program written in Occam-pi is set up like this:
```

Comment:
I prefer it when the text introduces the snippet. (i.e following shows a typical Occam-pi program"